### PR TITLE
Add custom erc20 contract to the EVM.

### PR DESCRIPTION
**Pull request by mfornet**
To compile the EVM it is required to compile ERC20 first if this change is used. Do this using the following steps:

```
cd eth-contracts
yarn
yarn compile
```

This will generate relevant binary in:
`eth-contracts/res/EvmErc20.bin`

### CODE REVIEW

**Comment by artob** [comment]
@mfornet Don't introduce new top-level directories, put `eth-contracts` under either `src` or `etc`.

**Comment by mfornet** [comment]
> @mfornet Don't introduce new top-level directories, put `eth-contracts` under either `src` or `etc`.

Sounds good to me. Actually I was thinking if we should even keep this solidity contract here or in separate repository and only pull binary.

I figured out it is better to have it here, since it is a small contract designed for be deployed as template by the evm.

Note that this includes those bytes in the evm binary, so the size will increase considerably.

**Comment by joshuajbouw** [comment]
Looks like CI needs to be updated to accommodate this?

**Comment by joshuajbouw** [comment]
> @mfornet Don't introduce new top-level directories, put `eth-contracts` under either `src` or `etc`.
I wouldn't put it under `src` as those are separate compiles, `etc` sounds better to me.

File: src/lib.rs [hunk 1]
```diff
@@ -262,8 +262,7 @@ mod contract {
 
         let mut engine = Engine::new(predecessor_address());
 
-        // TODO(#51): Use proper erc20_contract
-        let erc20_contract = Default::default();
+        let erc20_contract = include_bytes!("../eth-contracts/res/EvmErc20.bin");
```
Comment (joshuajbouw) [inline src/lib.rs:265]: I believe that our CI needs to be updated too for this?

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (mfornet) [reply src/lib.rs]: Done

**Comment by joshuajbouw** [comment]
Check tests

**Comment by joshuajbouw** [comment]
And lints :D

**Comment by birchmd** [comment]
I created an issue for myself to make use of this contract in the benchmarks after it is merged instead of the generic ERC20. We might as well measure the implementation we are actually going to use.

**Comment by artob** [comment]
@mfornet Looking forward to merging this. Please reassign to me once CI is showing green and it's ready to go.

**Comment by mfornet** [comment]
> @mfornet Looking forward to merging this. Please reassign to me once CI is showing green and it's ready to go.

@artob notice I'm merging this into #51 which is not ready.

**Comment by artob** [comment]
> And lints :D

Let's be clear that these precompiles are blockers for implementing and validating the connectors and hence blockers the release, and nice-to-haves like lints are a low priority right now.

**Review by joshuajbouw — APPROVED** [review]
Looks good.

**Pull request merged**